snippets/dart.json

```diff
@@ -0,0 +1,121 @@
+{
+	".source.dart": {
+		"main": {
+			"prefix": "main",
+			"body": [
+				"main(List<String> args) {",
+				"  $1",
+				"}"
+			]
+		},
+		"try": {
+			"prefix": "try",
+			"body": [
+				"try {",
+				"  $2",
+				"} catch (${1:e}) {",
+				"}"
+			]
+		},
+		"if": {
+			"prefix": "if",
+			"body": [
+				"if ($1) {",
+				"  $2",
+				"}"
+			]
+		},
+		"if else": {
+			"prefix": "ife",
+			"body": [
+				"if ($1) {",
+				"  $2",
+				"} else {",
+				"}"
+			]
+		},
+		"switch case": {
+			"prefix": "switch",
+			"body": [
+				"switch ($1) {",
+				"  case $2:",
+				"    $3",
+				"    break;",
+				"  default:",
+				"}"
+			]
+		},
+		"assert": {
+			"prefix": "assert",
+			"body": "assert($1);"
+		},
+		"print": {
+			"prefix": "print",
+			"body": "print($1);"
+		},
+		"for": {
+			"prefix": "for",
+			"body": [
+				"for (var i = 0; i < ${1:count}; i++) {",
+				"  $2",
+				"}"
+			]
+		},
+		"for in": {
+			"prefix": "fori",
+			"body": [
+				"for (var x in ${1:list}) {",
+				"  $2",
+				"}"
+			]
+		},
+		"while": {
+			"prefix": "while",
+			"body": [
+				"while ($1) {",
+				"  $2",
+				"}"
+			]
+		},
+		"do while": {
+			"prefix": "do",
+			"body": [
+				"do {",
+				"  $2",
+				"} while ($1);"
+			]
+		},
+		"void": {
```

Comment:
Many of these don't seem to work for me (`print`, `void`), because the completion provider is also providing completion for them and seems to take precedence, so I think maybe we should remove those ones? (I also think it's slightly confusing that sometimes you tab and get the semicolon on the end, and sometimes don't?